# Step: GO Enrichment Analysis on Differentially Expressed Variables

Perform GO enrichment analysis on differentially expressed variables
using
[`glystats::gly_enrich_go()`](https://glycoverse.github.io/glystats/reference/gly_enrich_go.html).
This step requires one of the DEA steps to be run. Only execute for
glycoproteomics experiments with exactly 2 groups. Use all genes in
OrgDb as the background.

## Usage

``` r
step_sig_enrich_go(universe = "all", ...)
```

## Arguments

- universe:

  The universe (background) to use for enrichment analysis. One of "all"
  (all genes in OrgDb), "detected" (detected variables in `exp`).

- ...:

  Step-specific arguments passed to underlying functions. Use the format
  `pkg.func.arg`. For example,
  `step_sig_enrich_go(glystats.gly_enrich_go.p_adj_method = "BH")`.

## Value

A `glysmith_step` object.

## Details

Data required:

- `exp`: The experiment to perform GO enrichment analysis for

- `sig_exp`: The filtered experiment from DEA, generated by
  `step_dea_xxx()`.

Tables generated:

- `go_enrich`: A table containing the GO enrichment results.

## Dynamic Arguments

This step supports the following dynamic arguments:

- `glystats.gly_enrich_go.OrgDb`: Organism database (default:
  "org.Hs.eg.db").

- `glystats.gly_enrich_go.readable`: Whether to map to readable gene
  names.

- `glyvis.plot_enrich.type`: Plot type ("dotplot", "barplot", etc.).

## See also

[`glystats::gly_enrich_go()`](https://glycoverse.github.io/glystats/reference/gly_enrich_go.html)

## Examples

``` r
step_sig_enrich_go()
#> <step "step_sig_enrich_go()"> GO enrichment analysis
```
