# Step: Reactome Enrichment Analysis on Differentially Expressed Variables

Perform Reactome enrichment analysis on differentially expressed
variables using
[`glystats::gly_enrich_reactome()`](https://glycoverse.github.io/glystats/reference/gly_enrich_go.html).
This step requires one of the DEA steps to be run. Only execute for
glycoproteomics experiments with exactly 2 groups. If used for glycomics
experiments, the step will be skipped. Use all genes in OrgDb as the
background.

## Usage

``` r
step_sig_enrich_reactome(universe = "all", ...)
```

## Arguments

- universe:

  The universe (background) to use for enrichment analysis. One of "all"
  (all genes in OrgDb), "detected" (detected variables in `exp`).

- ...:

  Step-specific arguments passed to underlying functions. Use the format
  `pkg.func.arg`. For example,
  `step_sig_enrich_reactome(glystats.gly_enrich_reactome.OrgDb = "org.Mm.eg.db")`.

## Value

A `glysmith_step` object.

## Details

Data required:

- `exp`: The experiment to perform Reactome enrichment analysis for

- `sig_exp`: The filtered experiment from DEA, generated by
  `step_dea_xxx()`.

Tables generated:

- `reactome_enrich`: A table containing the Reactome enrichment results.

## Dynamic Arguments

This step supports the following dynamic arguments:

- `glystats.gly_enrich_reactome.OrgDb`: Organism database (default:
  "org.Hs.eg.db").

- `glystats.gly_enrich_reactome.organism`: Organism name (default:
  "human").

- `glyvis.plot_enrich.type`: Plot type ("dotplot", "barplot", etc.).

## See also

[`glystats::gly_enrich_reactome()`](https://glycoverse.github.io/glystats/reference/gly_enrich_go.html)

## Examples

``` r
step_sig_enrich_reactome()
#> <step "step_sig_enrich_reactome()"> REACTOME enrichment analysis
```
