<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Get Started with glysmith • glysmith</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Get Started with glysmith">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">glysmith</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.8.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/glysmith.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/ai.html">AI-Powered Workflow in Glysmith</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Get Started with glysmith</h1>
            
      

      <div class="d-none name"><code>glysmith.Rmd</code></div>
    </div>

    
    
<p>Glycoverse is an incredibly comprehensive ecosystem. On one hand,
this means you can flexibly perform a wide range of analyses by
combining its different components. On the other hand, getting started
can require learning quite a lot.</p>
<p><code>glysmith</code> is the highest-level package in the glycoverse
suite. It encapsulates the most common analysis workflows into a unified
interface, allowing you to execute the entire analytical pipeline in
just one function call.</p>
<p>If you’ve seen The Lord of the Rings, you can think of
<code>glysmith</code> as the “One Ring to rule them all.”</p>
<p><em>Note: <code>glysmith</code> is not part of the core glycoverse
packages. You need to install it separately, even if you have installed
the meta package <code>glycoverse</code>.</em></p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://glycoverse.github.io/glysmith/">glysmith</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://glycoverse.github.io/glyexp/" class="external-link">glyexp</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="analyze-your-data-with-a-single-line-of-code">Analyze your data with a single line of code<a class="anchor" aria-label="anchor" href="#analyze-your-data-with-a-single-line-of-code"></a>
</h2>
<p>We will demonstrate using the built-in experiment object from
<code>glyexp</code>. This is an N-glycoproteomics dataset containing 12
samples across four different liver conditions: healthy controls (H),
hepatitis (M), cirrhosis (Y), and hepatocellular carcinoma (C), with 3
samples for each group. Please note, the dataset is raw and has not been
normalized or imputed.</p>
<p>To work with your own data, you can use <code>glyread</code> to
import outputs from tools like pGlyco3, MSFragger-Glyco, or Byonic;
Alternatively, you can create your own experiment object. For more
details, see <a href="https://glycoverse.github.io/glyread/articles/glyread.html" class="external-link">Get
Started with glyread</a> or <a href="https://glycoverse.github.io/glyexp/articles/create-exp.html" class="external-link">Creating
Experiments</a>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">real_experiment</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Glycoproteomics Experiment</span> <span style="color: #00BBBB;">──────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Expression matrix: <span style="color: #0000BB;">12</span> samples, <span style="color: #0000BB;">4262</span> variables</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Sample information fields: <span style="color: #00BB00;">group</span> <span style="color: #0000BB;">&lt;fct&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Variable information fields: <span style="color: #00BB00;">peptide</span> <span style="color: #0000BB;">&lt;chr&gt;</span>, <span style="color: #00BB00;">peptide_site</span> <span style="color: #0000BB;">&lt;int&gt;</span>, <span style="color: #00BB00;">protein</span> <span style="color: #0000BB;">&lt;chr&gt;</span>, <span style="color: #00BB00;">protein_site</span> <span style="color: #0000BB;">&lt;int&gt;</span>, <span style="color: #00BB00;">gene</span> <span style="color: #0000BB;">&lt;chr&gt;</span>, <span style="color: #00BB00;">glycan_composition</span> <span style="color: #0000BB;">&lt;glyrpr_c&gt;</span>, <span style="color: #00BB00;">glycan_structure</span> <span style="color: #0000BB;">&lt;glyrpr_s&gt;</span></span></span></code></pre></div>
<p>Now, let’s run the default analysis pipeline with just a single line
of code.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/forge_analysis.html">forge_analysis</a></span><span class="op">(</span><span class="va">real_experiment</span><span class="op">)</span></span>
<span><span class="va">result</span></span></code></pre></div>
<p>That’s it! You’ve completed the following steps in one go:</p>
<ul>
<li>Preprocessed your data using <code><a href="https://glycoverse.github.io/glyclean/reference/auto_clean.html" class="external-link">glyclean::auto_clean()</a></code> and
generated QC plots.</li>
<li>Summarized the experiment with
<code><a href="https://glycoverse.github.io/glyexp/reference/summarize_experiment.html" class="external-link">glyexp::summarize_experiment()</a></code>.</li>
<li>Performed principal component analysis with
<code><a href="https://glycoverse.github.io/glystats/reference/gly_pca.html" class="external-link">glystats::gly_pca()</a></code>.</li>
<li>Conducted differential expression analysis using
<code><a href="https://glycoverse.github.io/glystats/reference/gly_limma.html" class="external-link">glystats::gly_limma()</a></code>.</li>
<li>Plotted a volcano plot via <code><a href="https://glycoverse.github.io/glyvis/reference/plot_volcano.html" class="external-link">glyvis::plot_volcano()</a></code>.</li>
<li>Generated a heatmap for significant glycoforms using
<code><a href="https://glycoverse.github.io/glyvis/reference/plot_heatmap.html" class="external-link">glyvis::plot_heatmap()</a></code>.</li>
<li>Conducted GO enrichment analysis using
<code><a href="https://glycoverse.github.io/glystats/reference/gly_enrich_go.html" class="external-link">glystats::gly_enrich_go()</a></code>.</li>
<li>Performed KEGG enrichment analysis using
<code><a href="https://glycoverse.github.io/glystats/reference/gly_enrich_go.html" class="external-link">glystats::gly_enrich_kegg()</a></code>.</li>
<li>Performed Reactome enrichment analysis using
<code><a href="https://glycoverse.github.io/glystats/reference/gly_enrich_go.html" class="external-link">glystats::gly_enrich_reactome()</a></code>.</li>
<li>Derived site-specific traits using
<code><a href="https://glycoverse.github.io/glydet/reference/derive_traits.html" class="external-link">glydet::derive_traits()</a></code>.</li>
<li>Conducted differential trait analysis with
<code><a href="https://glycoverse.github.io/glystats/reference/gly_limma.html" class="external-link">glystats::gly_limma()</a></code>.</li>
<li>Generated a heatmap of significant site-specific derived traits via
<code><a href="https://glycoverse.github.io/glyvis/reference/plot_heatmap.html" class="external-link">glyvis::plot_heatmap()</a></code>.</li>
</ul>
<p>What a comprehensive analysis pipeline!</p>
</div>
<div class="section level2">
<h2 id="save-all-results-with-one-line-of-code">Save all results with one line of code<a class="anchor" aria-label="anchor" href="#save-all-results-with-one-line-of-code"></a>
</h2>
<p>Once you have your results, what’s next? You can export all tables
and plots to a target directory in just a single line of code.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/quench_result.html">quench_result</a></span><span class="op">(</span><span class="va">result</span>, <span class="st">"path/to/save"</span><span class="op">)</span></span></code></pre></div>
<p>Running this will create the folder <code>path/to/save</code> with
the following content:</p>
<ul>
<li>
<code>README.md</code>: Describes all the saved outputs.</li>
<li>
<code>experiment.rds</code>: The processed experiment object.</li>
<li>
<code>meta.rds</code>: The metadata list.</li>
<li>
<code>plots/</code>: All plots generated from the pipeline.</li>
<li>
<code>tables/</code>: All tables output from the analysis.</li>
</ul>
<p>Give it a try yourself!</p>
</div>
<div class="section level2">
<h2 id="one-line-to-generate-a-report">One line to generate a report<a class="anchor" aria-label="anchor" href="#one-line-to-generate-a-report"></a>
</h2>
<p>We’re developing advanced report generation features, but for now,
you can use <code><a href="../reference/polish_report.html">polish_report()</a></code> to generate a report directly
from the results.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/polish_report.html">polish_report</a></span><span class="op">(</span><span class="va">result</span>, <span class="st">"path/to/save/report.html"</span><span class="op">)</span></span></code></pre></div>
<p>By default, the report will open automatically in your browser when
it’s finished.</p>
</div>
<div class="section level2">
<h2 id="about-naming">About naming<a class="anchor" aria-label="anchor" href="#about-naming"></a>
</h2>
<p>You may have noticed some unusual names in this package. First, the
‘smith’ in <code>glysmith</code> doesn’t refer to a surname; it’s taken
from “blacksmith”—someone who makes and repairs items made of iron.</p>
<p>All functions and classes are named using blacksmith jargon. For
example, <code><a href="../reference/forge_analysis.html">forge_analysis()</a></code> for “forging” results,
<code><a href="../reference/quench_result.html">quench_result()</a></code> for “quenching” the results to save them,
and <code><a href="../reference/polish_report.html">polish_report()</a></code> for “polishing” the report to improve
readability.</p>
<p>You’ll learn about “blueprint” in the next section.</p>
</div>
<div class="section level2">
<h2 id="blueprints">Blueprints<a class="anchor" aria-label="anchor" href="#blueprints"></a>
</h2>
<p>Imagine you visit a blacksmith to order a sword. If you simply say,
“I want a sword,” the blacksmith will use a default blueprint and start
working on it. However, you always have the option to bring your own
design—a custom blueprint—that the smith can use for your sword.</p>
<p>Similarly, <code><a href="../reference/forge_analysis.html">forge_analysis()</a></code> uses a default blueprint for
the analysis pipeline. You can inspect it via
<code><a href="../reference/blueprint_default.html">blueprint_default()</a></code>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/blueprint_default.html">blueprint_default</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Blueprint (</span><span style="color: #0000BB; font-weight: bold;">13</span><span style="font-weight: bold;"> steps)</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; • step_ident_overview()</span></span>
<span><span class="co">#&gt; • step_preprocess()</span></span>
<span><span class="co">#&gt; • step_plot_qc(when = "post")</span></span>
<span><span class="co">#&gt; • step_pca()</span></span>
<span><span class="co">#&gt; • step_dea_limma()</span></span>
<span><span class="co">#&gt; • step_volcano()</span></span>
<span><span class="co">#&gt; • step_heatmap(on = "sig_exp")</span></span>
<span><span class="co">#&gt; • step_sig_enrich_go()</span></span>
<span><span class="co">#&gt; • step_sig_enrich_kegg()</span></span>
<span><span class="co">#&gt; • step_sig_enrich_reactome()</span></span>
<span><span class="co">#&gt; • step_derive_traits()</span></span>
<span><span class="co">#&gt; • step_dea_limma(on = "trait_exp")</span></span>
<span><span class="co">#&gt; • step_heatmap(on = "sig_trait_exp")</span></span></code></pre></div>
<p>A blueprint is a list of steps executed sequentially. You can use
<code><a href="../reference/blueprint.html">blueprint()</a></code> with <code>step_</code> functions to create
your own custom blueprint. For instance, if you’d just like to perform
differential expression analysis and plot the volcano plot, you could
do:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/blueprint.html">blueprint</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="../reference/step_preprocess.html">step_preprocess</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="../reference/step_dea_limma.html">step_dea_limma</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="../reference/step_volcano.html">step_volcano</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span><span class="op">)</span></span>
<span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/forge_analysis.html">forge_analysis</a></span><span class="op">(</span><span class="va">real_experiment</span>, <span class="va">bp</span><span class="op">)</span></span>
<span><span class="va">result</span></span></code></pre></div>
<p>You can also save a blueprint to a file and reload it later:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/write_blueprint.html">write_blueprint</a></span><span class="op">(</span><span class="va">bp</span>, <span class="st">"path/to/save/bp.rds"</span><span class="op">)</span></span>
<span><span class="va">bp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/write_blueprint.html">read_blueprint</a></span><span class="op">(</span><span class="st">"path/to/save/bp.rds"</span><span class="op">)</span></span></code></pre></div>
<p>Here is a list of all available steps:</p>
<ul>
<li>
<code><a href="../reference/step_ident_overview.html">step_ident_overview()</a></code>: Summarize the experiment
identification.</li>
<li>
<code><a href="../reference/step_preprocess.html">step_preprocess()</a></code>: Preprocess the experiment data and
generate QC plots.</li>
<li>
<code><a href="../reference/step_pca.html">step_pca()</a></code>, <code><a href="../reference/step_tsne.html">step_tsne()</a></code>,
<code><a href="../reference/step_umap.html">step_umap()</a></code>: Dimension reduction for data exploration.</li>
<li>
<code><a href="../reference/step_dea_limma.html">step_dea_limma()</a></code>, <code><a href="../reference/step_dea_ttest.html">step_dea_ttest()</a></code>,
<code><a href="../reference/step_dea_anova.html">step_dea_anova()</a></code>, <code><a href="../reference/step_dea_wilcox.html">step_dea_wilcox()</a></code>,
<code><a href="../reference/step_dea_kruskal.html">step_dea_kruskal()</a></code>: Differential expression analysis with
various methods.</li>
<li>
<code><a href="../reference/step_volcano.html">step_volcano()</a></code>: Create volcano plots for DEA
results.</li>
<li>
<code><a href="../reference/step_heatmap.html">step_heatmap()</a></code>: Generate heatmaps for expression
data.</li>
<li>
<code><a href="../reference/step_roc.html">step_roc()</a></code>: Perform Receiver Operating Characteristic
(ROC) analysis.</li>
<li>
<code><a href="../reference/step_sig_enrich_go.html">step_sig_enrich_go()</a></code>,
<code><a href="../reference/step_sig_enrich_kegg.html">step_sig_enrich_kegg()</a></code>,
<code><a href="../reference/step_sig_enrich_reactome.html">step_sig_enrich_reactome()</a></code>: Functional enrichment analysis
for significant items.</li>
<li>
<code><a href="../reference/step_derive_traits.html">step_derive_traits()</a></code>: Calculate glycan derived
traits.</li>
<li>
<code><a href="../reference/step_quantify_motifs.html">step_quantify_motifs()</a></code>: Quantify glycan motifs
(substructures).</li>
</ul>
<p>Before you start building your own blueprints, here are a few rules
to keep in mind:</p>
<ul>
<li>
<code><a href="../reference/step_preprocess.html">step_preprocess()</a></code> is generally the first step, but it’s
optional if you perform preprocessing manually with
<code>glyclean</code>.</li>
<li>Some steps require others to be executed first. For example,
<code><a href="../reference/step_volcano.html">step_volcano()</a></code> requires at least one of
<code><a href="../reference/step_dea_limma.html">step_dea_limma()</a></code>, <code><a href="../reference/step_dea_ttest.html">step_dea_ttest()</a></code>, or
<code><a href="../reference/step_dea_wilcox.html">step_dea_wilcox()</a></code> to be run beforehand. Consult the
documentation for each step to check dependencies.</li>
<li>You cannot include duplicate steps within a blueprint. For instance,
you can’t add <code><a href="../reference/step_dea_limma.html">step_dea_limma()</a></code> twice in the same
blueprint.</li>
<li>If a step overwrites data produced by an earlier step, a warning
will be issued. For example, including both
<code><a href="../reference/step_dea_limma.html">step_dea_limma()</a></code> and then <code><a href="../reference/step_dea_ttest.html">step_dea_ttest()</a></code>
will overwrite the <code>dea_res</code> data from the former with that
from the latter step. Ignore the warning if that’s what you intend.</li>
<li>You can create branches in a blueprint by using <code><a href="../reference/br.html">br()</a></code>.
Check out its document for examples.</li>
</ul>
<p><strong>Happy forging!</strong></p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Bin Fu.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
